.github/workflows/reusable-build-docs.yaml

```
@@ -50,26 +47,34 @@ jobs:
           }}
         ref: ${{ inputs.repository-branch }}
         path: build-directory
+        persist-credentials: false
 
     - name: Setup nox
       uses: wntrblm/nox@2025.10.16
 
     - name: Output Python info
       run: python --version --version && which python
 
-    - name: Set the VERSION variable
-      run: echo VERSION="${PACKAGE_VERSION}" >> "${GITHUB_ENV}"
+    - name: Graft ansible-core
+      run: nox -s clone-core
+      working-directory: build-directory
 
     - name: Build the Ansible community package docs
-      run: >-
-        nox -e make -- webdocs ${{
+      env:
+        PACKAGE_VERSION: "${{ inputs.ansible-package-version }}"
```

@oraNod, do we need to also export VERSION like it was before in the `Set the VERSION variable` step? Is `$VERSION` used in the Makefile?

```suggestion
        PACKAGE_VERSION: "${{ inputs.ansible-package-version }}"
        VERSION: "${{ inputs.ansible-package-version }}"
```

---

Hi @gotmax23 `ANSIBLE_VERSION` gets passed to the Makefile but not `VERSION`.

I think `VERSION` is a relic from the Jenkins job that is used in a couple of places including the step to add a symlink on the web server: `ln -s \$VERSION _build/latest`. Maybe we needed it when we first started with the workflow but it should be fine to drop it as part of this PR.

I'm pretty sure I built this branch before I approved but will do it again now. Thanks, as always, for the thoughtful questions.

---

Just did a run with this branch from the devel branch and see this in the logs: `nox > make -C docs/docsite PYTHON=/home/runner/work/ansible-documentation/ansible-documentation/build-directory/.nox/make/bin/python webdocs ANSIBLE_VERSION= 'EXTRA_TAGS=-t redirects'`

Everything in the tarball looks good too. I think we can safely remove the `Set the VERSION variable` step.

---

I see

``` console
$ rg '^VERSION.*=' docs/docsite/Makefile
VERSION := $(shell $(PYTHON) ./version_helper.py --raw || echo error)
```
. Assuming I remember how Makefile syntax works, I think this will set VERSION to a default value unless it is already set. `VERSION` was previously set in the environment, but now it's not, so I think that the value generated by that script would be used instead. I'm not sure what impact that has.

---

Hmm. I thought that line was more related to how the core versions get set in docs. But it's true that `VERSION` no longer gets set in the environment.

If there's any impact it would be to the version list drop down, which I can see is still at `11` here: https://docs.ansible.com/projects/ansible/devel/index.html

Perhaps we stomped on how `VERSION` was getting set when we made this a reusable workflow and it's been broken? I'll go back and look. I've compared the generated HTML built from this branch with what we're building on devel and they seem to match. So I really don't think we're going to break something new by dropping `VERSION`.

I'm also not sure it's worth fixing whatever `VERSION` was doing for the package docs. We've got the Read The Docs flyout menu on `docs.ansible.com` now. Part of the migration effort was to adopt RTD native tooling instead of the custom version switcher. That might be a good step towards decoupling package and core docs, which is also the root of some of the complexity in the Makefile.

---

@gotmax23 Hey, I've looked into this a little more. The `:=` operator there ensures that the value of `VERSION` is always set by the core version helper script. I found [this page](https://earthly.dev/blog/makefile-variables/) that describes how that variable works.

I also tested with a simple test file that contains:

```
VERSION := "1234"

show:
	@echo "VERSION = $(VERSION)"
```

Then tested the behavior:

```
$ make -f test.mk 
VERSION = 1234

$ VERSION=678 make -f test.mk show
VERSION = 1234
```

If you then change to a conditional variable `VERSION ?= "1234"` and run the same test the value from the environment is used.

Again, I'm not an expert on the syntax. The Makefile has a pretty long history and others probably know better. I am really confident that we don't need to set `VERSION` in the environment as part of the docs build though.

If you think it's better, I can remove that step in a separate PR. That might be a cleaner approach.

---

Ah, okay thanks for checking @oraNod.

Resolving...